Tella/Scenes/Audio/AudioPlayer/Views/EditAudioView.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  EditAudioView.swift
+//  Tella
+//
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftUI
+
+
+
+struct EditAudioView: View {
+    @EnvironmentObject var sheetManager: SheetManager
+    
+    @StateObject var editAudioViewModel: EditAudioViewModel
+    @Binding var isPresented : Bool
+    @State var isBottomSheetShown : Bool = false
+    let kTrimViewWidth = 340.0
+    
+    @State var trailingGestureValue: Double = 0.0
+    @State var leadingGestureValue: Double = 0.0
+    @State var shouldStopLeftScroll = false
+    @State var shouldStopRightScroll = false
+    
+    var body: some View {
+        ZStack {
+            VStack {
+                headerView
+                timeLabelsView
+                trimView
+                displayTimeLabels
+                controlButtonsView
+                Spacer()
+            }
+            EditFileCancelBottomSheet(isShown: $isBottomSheetShown, saveAction: { editAudioViewModel.trimAudio() })
+            if case .error(let message) = editAudioViewModel.trimState{
+                ToastView(message: message)
+            }
+        }
+        .onAppear {
+            editAudioViewModel.onAppear()
+            trailingGestureValue = kTrimViewWidth
+        }
+        .onReceive(editAudioViewModel.$trimState) { value in
+            if value == .loaded(true) {
```

Comment:
What do you think of adding a handleTrimState function `handleTrimState` that contains the .error/.loaded states 


    func handleTrimState(value:ViewModelState<Bool>) {
       
        switch value {
        case .loaded(let isLoaded):
            if isLoaded {
                self.isPresented = false
                self.sheetManager.hide()
                Toast.displayToast(message: LocalizableVault.editFileSavedToast.localized)
            }
        case .error(let message):
            Toast.displayToast(message: message)

        default:
            break
        }
    }